### ISSUE TITLE
Compilation errors for math_gpu.cu

### ISSUE THREAD

**Issue by sherifshehata** [facebookarchive/caffe2#331]
I am building on Ubuntu 14.04 using cuda 7.5. I get these errors.
(p.s. I pulled the latest code today, yesterday i didn't get these errors.)

> [ 73%] Building NVCC (Device) object caffe2/CMakeFiles/Caffe2_GPU.dir/sgd/./Caffe2_GPU_generated_adagrad_op_gpu.cu.o
/home/sherif/caffe2/caffe2/utils/conversions.h(137): warning: variable "r" was declared but never referenced

> /home/sherif/caffe2/caffe2/utils/math_gpu.cu(136): error: identifier "CUDA_R_16F" is undefined

> /home/sherif/caffe2/caffe2/utils/math_gpu.cu(312): error: identifier "CUDA_R_16F" is undefined

> /home/sherif/caffe2/caffe2/utils/math_gpu.cu(496): error: identifier "CUDA_R_16F" is undefined

> /home/sherif/caffe2/caffe2/utils/math_gpu.cu(496): error: identifier "CUDA_R_32F" is undefined

> /home/sherif/caffe2/caffe2/utils/math_gpu.cu(496): error: identifier "cublasDotEx" is undefined

> /home/sherif/caffe2/caffe2/utils/math_gpu.cu(740): error: identifier "CUDA_R_16F" is undefined

> /home/sherif/caffe2/caffe2/utils/math_gpu.cu(740): error: identifier "CUDA_R_32F" is undefined

> /home/sherif/caffe2/caffe2/utils/math_gpu.cu(740): error: identifier "cublasAxpyEx" is undefined

> 8 errors detected in the compilation of "/tmp/tmpxft_00004bd7_00000000-7_math_gpu.cpp1.ii".
CMake Error at Caffe2_GPU_generated_math_gpu.cu.o.cmake:260 (message):
  Error generating file
  

>/home/sherif/caffe2/build/caffe2/CMakeFiles/Caffe2_GPU.dir/utils/./Caffe2_GPU_generated_math_gpu.cu.o

**Comment by slayton58** [comment]
These descriptors were added in CUDA 8.0 - I suggest upgrading to that version

**Comment by snehashis-roy** [comment]
I also got the same errors with cuda 7.5. But when I used cuda 8.0,  I got the following error. My driver version is 361.28 though.

Could NOT find CUDA: Found unsuitable version "8.0", but required is exact version "7.5" (found /home/piby2/cuda-8.0)

**Comment by sherifshehata** [comment]
Thank you. The installation​guide doesn't say that cuda 8 is required, it says anything higher than 6.5 is ok. I think you should update the requirements in the documentation.

**Comment by snehashis-roy** [comment]
Did you get it successfully compiled with cuda 8.0?

**Comment by Yangqing** [comment]
Yeah, I think we should suggest cuda 8 now - @slayton58 would that make sense?

**Comment by slayton58** [comment]
Yes, requiring CUDA >= 8.0 would make sense. 

This was a very recent addition (related to fp16 operators) which is why docs haven't quite caught up :)

**Comment by snehashis-roy** [comment]
I am getting this error with 8.0, 
Could NOT find CUDA: Found unsuitable version "8.0", but required is exact version "7.5" (found /home/piby2/cuda-8.0)
I have nvidia 361.28 drivers, cuda 8.0, and cudnn 5.0.5.

Did anybody  else get this error? @slayton58 @Yangqing 
Thank you.

**Comment by slayton58** [comment]
That's very strange - we require 7.0 and above in the main cmake scripts, so 8.0 should be perfectly fine.

I wonder if your low driver is confusing someone somewhere - I don't believe that's the 8.0 release driver version, so if you could upgrade to that (not sure exactly what that is offhand) and try again, that'd be appreciated!

**Comment by alexge233** [comment]
I am getting this error with CUDA 7.5 on Debian 8.0. I am guessing this is related to half floats?

**Comment by slayton58** [comment]
Yes, this comes from fp16 support, which requires the `CUDA_R_*F` identifiers
